orm/orm.go

```diff
@@ -157,13 +322,23 @@ func (o *orm) ReadOrCreate(md interface{}, col1 string, cols ...string) (bool, i
 // insert model data to database
 func (o *orm) Insert(md interface{}) (int64, error) {
 	mi, ind := o.getMiInd(md, true)
+	name := mi.fullName
+	if hookFunc, ok := o.beforeInsert[name]; ok {
+		hookErr := hookFunc(md)
+		if hookErr != nil {
+			return 0, hookErr
+		}
+	}
 	id, err := o.alias.DbBaser.Insert(o.db, mi, ind, o.alias.TZ)
 	if err != nil {
 		return id, err
 	}
-
 	o.setPk(mi, ind, id)
-
+	if hookFunc, ok := o.afterInsert[name]; ok {
+		go func(md interface{}) {
```

Comment:
You can't use goroutine for these functions. As it may cause the data not updated.

Reply:
Using goroutine only if there is 'after' hook function, 'before' hook will not use goroutine.

Reply:
but if after hook failed, what can we do?

Reply:
sorry i don't follow, 'after' hook failed can cause the same as other function failed, it'll not cause the whole 'CRUD' function failed.
maybe need a recover() here? thought it's better to let recover() called by owner of hook().

Reply:
I just discuss with the gorm author, in gorm any hooks failed will rollback this action

Reply:
sorry for delay.
after study gorm's callback method, i found that gorm using tx(transaction) as param of ['afterInsert', 'afterUpdate', 'afterDelete'] to achieve db rollback, means in those hook functions, we still can't use the 'result' of a CRUD function, that's not an 'after' hook to me, because an 'after' function should let we use an 'updated' object(result after object be updated) right? those 'after' callbacks can all put into current tx, because they using current tx, sounds like an 'append' to me. so maybe they also need a PR :)
My thoughts:
'After' hook should let us produce the 'after-transaction' object.
In this way, 'after' hook still can failed, but no need to add db rollback in it. so maybe no goroutine for 'after' hook, let developers to choose whether goroutine or not?

Reply:
Maybe 'after' hook need goroutine after all, avoid 'after' hook failed causing the whole CRUD function failed.

Reply:
I think the after hook just return error, but no panic

Reply:
Here have some ideas https://github.com/ilibs/gosql#hooks

Reply:
a bit weird this way, CRUD function return error but with correct influenced lines count. 
and add hook functions into object like gorm maybe more clearly?
`func (s *Product) AfterCreate()`